### PR TITLE
Remaining translation, except diagrams

### PR DESCRIPTION
**Pull request by adamAndMath**
I've used 2 libraries whose files are currently just included:
- https://github.com/sahasatvik/typst-theorems
- https://gitlab.com/giacomogallina/typst-cd

I modified typst-cd to not straight up fail on loops. However the edges are still wrong. Further more, I've left out the higher order diagrams, as I couldn't get them to work.

### CODE REVIEW

File: src/definition.typ [hunk 1]
```diff
@@ -0,0 +1,85 @@
+#import "lib.typ": *
+#import "typst-cd.typ": node, arr, commutative_diagram
+
+= Definition of a category, and internal properties
+
+#definition[
+    (Category). A category is the followin things:
+    - A collection of objects,
+    - A collection of morphism, or arraws, such that each morphism $f$ has a _domain_ and _codomain_. A morphism $f$ of domain $x$ and codomain $y$ will be written $f : x -> y$. We also ask the following properties:
+        - for $f : x -> y$ and $g : y -> z$, there is a morphism $g f : x -> z$. Two such morphisms $f$ and $g$ will be called _composable_,
```
Comment (lafeychine) [inline src/definition.typ:10]: Is this change was a mistake in the original document?
```suggestion
        - for $f : x -> y$ and $g : y -> z$, there is a morphism $f g : x -> z$. Two such morphisms $f$ and $g$ will be called _composable_,
```

File: src/definition.typ [hunk 2]
```diff
@@ -0,0 +1,85 @@
+#import "lib.typ": *
+#import "typst-cd.typ": node, arr, commutative_diagram
+
+= Definition of a category, and internal properties
+
+#definition[
+    (Category). A category is the followin things:
```
Comment (lafeychine) [inline src/definition.typ:7]:
```suggestion
#definition(name: "Category")[
    A category is the following things:
```

File: src/definition.typ [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    - A collection of objects,
+    - A collection of morphism, or arraws, such that each morphism $f$ has a _domain_ and _codomain_. A morphism $f$ of domain $x$ and codomain $y$ will be written $f : x -> y$. We also ask the following properties:
```
Comment (lafeychine) [inline src/definition.typ:9]:
```suggestion
    - A collection of morphism, or arrows, such that each morphism $f$ has a _domain_ and _codomain_. A morphism $f$ of domain $x$ and codomain $y$ will be written $f : x -> y$. We also ask the following properties:
```

File: src/definition.typ [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        - for $f : x -> y$ and $g : y -> z$, there is a morphism $g f : x -> z$. Two such morphisms $f$ and $g$ will be called _composable_,
+        - for each object $x$ there is a morphism $1_x : x -> x$,
```
Comment (lafeychine) [inline src/definition.typ:11]: Is this change was a mistake in the original document?
```suggestion
        - for each element $x$ there is a morphism $1_x : x -> x$,
```

**Comment by lafeychine** [comment]
This MR was originally intended to be just a portage from LaTeX to Typst, but this MR also contains changes.

This makes reviewing `first_part` more difficult, as we cannot discuss the word/sentence you want to change.

I will close this issue and backport any changes you made in the original MR review.

Closing.

File: src/definition.typ (hunk 1 above)
Comment (adamAndMath) [reply src/definition.typ:10]: Yes. Composition is should be backwards (like I wrote). This is also how it was used in the rest of the Tex.

File: src/definition.typ (hunk 4 above)
Comment (adamAndMath) [reply src/definition.typ:11]: I feel it's clearer, as elements refer to sets or collections. As categories contain multiple collections, I feel it's clearer to specify which one.

**Comment by lafeychine** [comment]
Your work has been being merged into `first_part` branch, with the following changes:
 - Use of `cat(C)` and `cat(D)` instead of the confusing `cat` and `cat2`
 - Make as "TODO" diagrams we can't write yet with typst-cd
 - Import dependencies as git submodules

**Pull request closed without merging**